bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/AbstractTextEditor.java

```diff
@@ -7095,6 +7114,38 @@ protected void doRestoreState(IMemento memento) {
 			textWidget.setHorizontalPixel(horizontalPixel.intValue());
 	}
 
+	@SuppressWarnings("boxing")
+	private ISelection readSelectionFromMemento(IMemento memento) {
+		if (memento.getBoolean(TAG_IS_MULTISELECTION)) {
+			int selectionRegions = memento.getInteger(TAG_MULTISELECTION_COUNT);
+			Vector<IRegion> regions = new Vector<>();
+			for (int i = 0; i < selectionRegions; i++) {
+				Integer offset = memento.getInteger(TAG_SELECTION_OFFSET + i);
+				if (offset == null) {
+					continue;
+				}
+
+				Integer length = memento.getInteger(TAG_SELECTION_LENGTH + i);
+				if (length == null) {
+					continue;
+				}
+				regions.add(new Region(offset, length));
+			}
+			return new MultiTextSelection(new Document(), regions.toArray(new IRegion[0]));
```

Comment:
Is using an empty document for the `MultiTextSelection` permissible in this case? If not, which document should I pass to it?

Reply:
No, it should be something like `getDocumentProvider().getDocument(getEditorInput())`